crates/sui/src/client_commands.rs

```diff
@@ -1359,6 +1365,52 @@ impl Display for SuiClientCommandResult {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         let mut writer = String::new();
         match self {
+            SuiClientCommandResult::Addresses(addresses) => {
+                let json_obj = json!(addresses);
+                let mut table = json_to_table(&json_obj);
+                let style = TableStyle::rounded().horizontals([]);
+                table.with(style);
+                write!(f, "{}", table)?
+            }
+            SuiClientCommandResult::DynamicFieldQuery(df_refs) => {
+                let mut builder = TableBuilder::default();
+                builder.set_header([
+                    "name",
+                    "type",
+                    "objectType",
+                    "objectId",
+                    "version",
+                    "digest",
+                ]);
+
+                for df_ref in df_refs.data.iter() {
+                    let df_type = match df_ref.type_ {
+                        DynamicFieldType::DynamicField => "DynamicField",
+                        DynamicFieldType::DynamicObject => "DynamicObject",
+                    };
+                    builder.push_record([
+                        df_ref.name.to_string(),
+                        df_type.to_string(),
+                        df_ref.object_type.to_string(),
+                        df_ref.object_id.to_string(),
+                        df_ref.version.value().to_string(),
+                        Base64::encode(df_ref.digest),
+                    ]);
+                }
+                let mut table = builder.build();
+                let info_header = if let Some(cursor) = df_refs.next_cursor {
+                    format!(
+                        "Next cursor: {cursor}. Showing {} elements",
+                        df_refs.data.len()
```

Comment:
nit: When I read this, it makes me think there are that many elements in the next cursor, rather than that there are that many elements so far and more to come in the next cursor.  Maybe swap these two messages around?

```suggestion
                        "Showing {} elements. Next cursor: {cursor}.",
                        df_refs.data.len()
```